internal/catalogmetadata/client/client.go

```diff
@@ -12,15 +16,31 @@ import (
 	"github.com/operator-framework/operator-controller/internal/catalogmetadata"
 )
 
-func New(cl client.Client) *Client {
-	return &Client{cl: cl}
+// Fetcher is an interface to facilitate fetching
+// catalog contents from catalogd.
+type Fetcher interface {
+	// FetchCatalogContents fetches contents from the catalogd HTTP
+	// server for the catalog provided. It returns an io.ReadCloser
+	// containing the FBC contents that the caller is expected to close.
+	// returns an error if any occur.
+	FetchCatalogContents(ctx context.Context, catalog *catalogd.Catalog) (io.ReadCloser, error)
+}
+
+func New(cl client.Client, fetcher Fetcher) *Client {
```

Comment:
I called this thing "client" due to lack of imagination on my side. Don't hesitate to rename, if you feel like it & have a better name in mind.